base/sparse/cholmod.jl

```diff
@@ -87,22 +87,22 @@ function __init__()
             tmp = Array(Cint, 3)
             ccall((:cholmod_version, :libcholmod), Cint, (Ptr{Cint},), version_array)
             ccall((:jl_cholmod_version, :libsuitesparse_wrapper), Cint, (Ptr{Cint},), tmp)
-            if tmp != version_array
+            if tmp[1] != version_array[1]
```

Comment:
Ah, forgot to say I had checked the changelog before making this PR. I think we need to separate two things, which are basically what every sane library versioning policy follows:
1) Julia cannot use a different cholmod major version than the one it was built with because of ABI compatibility
2) Julia also has a minimum version requirement: it needs at least the first version in which all the used functions are present

1) is basically what the linker gives you automatically in C code (requiring the major SOVERSION). This is what this PR does. 2) is what is always handled separately by `configure` scripts, by package managers, or manually in Makefiles: just check that the version is above the minimum; most C programs don't check anything on their own though, and will just crash when they encounter an undefined symbol.

Of course we can do better than that. I can add another commit in which I check that the version is beyond a minimum. I can also have a look to find out what version that corresponds to.